sirepo/package_data/static/js/controls.js

```diff
@@ -314,37 +320,80 @@ SIREPO.app.controller('ControlsController', function(appState, controlsService,
         }
     }
 
+    function loadTwissReport(data) {
+        appState.models.instrumentAnimationTwiss.refreshId = Math.random();
+        appState.models.instrumentAnimationTwiss.valueList = {
+            x: data.twissColumns,
+            y1: data.twissColumns,
+            y2: data.twissColumns,
+            y3: data.twissColumns,
+        };
+        self.twissReport = [{
+            modelKey: 'instrumentAnimationTwiss',
+            getData: genGetDataFunction('instrumentAnimationTwiss')
+        }];
+        frameCache.setFrameCount(1, 'instrumentAnimationTwiss');
+        appState.saveChanges('instrumentAnimationTwiss');
+    }
+
     function genGetDataFunction(m) {
         return () => appState.models[m];
     }
 
-    function initInstrumentModels() {
-        for (const m in appState.models) {
-            if (m.includes('instrumentAnimation')) {
-                return;
-            }
+    function initInstruments() {
+        if (checkModelSet('instrumentAnimation')){
+            return;
         }
         const k  = [];
+        const modelViewName = 'instrumentAnimation';
         appState.models.externalLattice.models.elements.forEach((e, i) => {
                 if (e.type !== 'INSTRUMENT') {
                     return;
                 }
-                const m = 'instrumentAnimation' + i;
+                const m = modelViewName + i;
                 k.push(m);
-                const n = {
-                    id: i,
-                };
-                appState.setModelDefaults(n, 'instrumentAnimation');
-                appState.models[m] = n;
+                setAnimationModel(m, modelViewName, i);
         });
         appState.saveChanges(k);
     }
 
+    function setAnimationModel(modelKey, modelViewName, id) {
+        const n = {
+            id: id,
+            viewName: modelViewName,
+        };
+        appState.setModelDefaults(n, modelViewName);
+        appState.models[modelKey] = n;
+    }
+
+    function initTwiss() {
+        if (checkModelSet('instrumentAnimationTwiss')){
+            return;
+        }
+        const keys = [];
+        const modelKey = 'instrumentAnimationTwiss';
+        const modelViewName = 'instrumentAnimationTwiss';
+        const id = appState.models.externalLattice.models.length + 2;
+        keys.push(modelKey);
+        setAnimationModel(modelKey, modelViewName, id);
+        appState.saveChanges(keys);
+    }
```

Comment:
I think all the "const xxx" values should be removed and placed within the calls. ex. setAnimationModel('instrumentAnimationTiwss', 'instrumentAnimationTwiss', appState.models.externalLattice.models.length + 2);